core/services/gateway/handlers/functions/handler.functions.go

```diff
@@ -111,7 +111,12 @@ func NewFunctionsHandlerFromConfig(handlerConfig json.RawMessage, donConfig *con
 		if err2 != nil {
 			return nil, err2
 		}
-		allowlist, err2 = NewOnchainAllowlist(chain.Client(), *cfg.OnchainAllowlist, lggr)
+
+		orm, err2 := NewORM(db, lggr, qcfg, cfg.OnchainAllowlist.ContractAddress)
```

Comment:
Why do we have to create 2 different ORM instances? I see we also have one on line 144.

Reply:
we are storing the router contract address as part of the key (both on allowlist and subscriptions).
I've seen in the code that on allowlist, the router address is extracted from `cfg.OnchainAllowlist.ContractAddress`
while on subscriptions, the router address is extracted from `cfg.OnchainSubscriptions.ContractAddress`

I might have misunderstood or missed something, happy to change it if there's a way of having only one

Reply:
Gotcha. Let me think about this.

Reply:
Yeah, I think 2 separate ORM instances works. I don't think there is any harm in doing so anyway.